protocols/gossipsub/src/behaviour.rs

```diff
@@ -1758,31 +1754,6 @@ where
             metrics.msg_recvd_unfiltered(&raw_message.topic, raw_message.raw_protobuf_len());
         }
 
-        let fast_message_id = self.config.fast_message_id(&raw_message);
-
-        if let Some(fast_message_id) = fast_message_id.as_ref() {
-            if let Some(msg_id) = self.fast_message_id_cache.get(fast_message_id) {
-                let msg_id = msg_id.clone();
-                // Report the duplicate
-                if self.message_is_valid(&msg_id, &mut raw_message, propagation_source) {
-                    if let Some((peer_score, ..)) = &mut self.peer_score {
-                        peer_score.duplicated_message(
-                            propagation_source,
-                            &msg_id,
-                            &raw_message.topic,
-                        );
-                    }
-                    // Update the cache, informing that we have received a duplicate from another peer.
-                    // The peers in this cache are used to prevent us forwarding redundant messages onto
-                    // these peers.
-                    self.mcache.observe_duplicate(&msg_id, propagation_source);
-                }
-
-                // This message has been seen previously. Ignore it
-                return;
-            }
-        }
-
         // Try and perform the data transform to the message. If it fails, consider it invalid.
         let message = match self.data_transform.inbound_transform(raw_message.clone()) {
```

Comment:
On failure, we call `self.handle_invalid_message` which as well will call `self.data_transform.inbound_transform`. Is that an issue @AgeManning?

Reply:
Yeah, its fine, but maybe in a future PR we can optimize it to prevent trying it twice. 

Because the decompression/transformation algo is user-specified, it could open DOS vectors if this transformation is very time-consuming by making the node run it twice through sending various messages that cause the algo to do expensive calcs. 

I dont think it's a concern for me specifically tho.